packages/edge/android/src/main/java/com/adobe/marketing/mobile/reactnative/edge/RCTAEPEdgeDataBridge.java

```diff
@@ -0,0 +1,107 @@
+
+import android.util.Log;
+
+import com.adobe.marketing.mobile.EdgeEventHandle;
+import com.adobe.marketing.mobile.ExperienceEvent;
+import com.facebook.react.bridge.ReadableArray;
+import com.facebook.react.bridge.ReadableMap;
+import com.facebook.react.bridge.ReadableType;
+import com.facebook.react.bridge.WritableMap;
+import com.facebook.react.bridge.WritableNativeMap;
+import com.facebook.react.bridge.WritableArray;
+
+import java.util.List;
+import java.util.Map;
+
+public final class RCTAEPEdgeDataBridge {
+
+    public final static String XDM_DATA_KEY = "xdmData";
+    public final static String DATA_KEY = "data";
+    public final static String DATASET_IDENTIFIER_KEY = "datasetIdentifier";
+
+    public final static String TYPE_KEY = "type";
+    public final static String PAYLOAD_KEY = "payload";
+    private static final String TAG = "RCTAEPEdgeDataBridge";
+
+    /**
+     * Converts a {@link ReadableMap} into an {@link ExperienceEvent}
+     *
+     * @param map
+     * @return An {@link ExperienceEvent}
+     */
+    public static ExperienceEvent experienceEventFromReadableMap(final ReadableMap map) {
+        if (map == null) {
+            return null;
+        }
+
+        Map<String, Object> xdmdata = RCTAEPEdgeMapUtil.toMap(getNullableMap(map, XDM_DATA_KEY));
+        String datasetId = null;
+
+        if (xdmdata != null) {
+
+            Map<String, Object> data = RCTAEPEdgeMapUtil.toMap(getNullableMap(map, DATA_KEY));
+
+            try {
+                datasetId = getNullableString(map, DATASET_IDENTIFIER_KEY);
+            } catch (Exception e) {
+                Log.d(TAG, "experienceEventFromReadableMap: " + e);
+            }
+
+            if (!(data instanceof Map)) {
+                Log.d(TAG, "experienceEventFromReadableMap: experience event free form data is null");
+            }
+
+            ExperienceEvent event = new ExperienceEvent.Builder().setXdmSchema(xdmdata, datasetId).setData(data).build();
+
+            return event;
+        }
+
+        Log.d(TAG, "experienceEventFromReadableMap: xdmdata is required, but it is currently null.");
+        return null;
+    }
+
+    /**
+     * Converts a {@link EdgeEventHandle} into a {@link WritableMap}
+     * @param eventhandle
+     * @return A {@link WritableMap} that represents the eventhandle
+     */
+    public static WritableMap mapFromEdgeEventHandle(final EdgeEventHandle eventhandle) {
+        if (eventhandle == null) {
+            return null;
+        }
+
+        if (eventhandle.getPayload() != null) {
+            Object[] handles = new Object[] {eventhandle.getPayload().size()};
+            handles = eventhandle.getPayload().toArray();
+
+            WritableMap eventHandleMap = new WritableNativeMap();
+            eventHandleMap.putString(TYPE_KEY, eventhandle.getType());
+            eventHandleMap.putArray(PAYLOAD_KEY, RCTAEPEdgeArrayUtil.toWritableArray(handles));
+
+            return eventHandleMap;
+        }
```

Comment:
We should read type and payload independently, in case the event handle has only one or the other. For example:
```
WritableMap eventHandleMap = new WritableNativeMap();
if (eventhandle.getType() != null) {
    eventHandleMap.putString(TYPE_KEY, eventhandle.getType());
}
if (eventhandle.getPayload() != null) {
    Object[] handles = new Object[] {eventhandle.getPayload().size()};
    handles = eventhandle.getPayload().toArray();
    eventHandleMap.putArray(PAYLOAD_KEY, RCTAEPEdgeArrayUtil.toWritableArray(handles));
}
return eventHandleMap;
```